spec/classes/minecraft_source_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe 'minecraft::source', :type => :class do
+  let :pre_condition do
+    'include ::minecraft'
+  end
+
+  let :facts do
+    {
+      :osfamily               => 'debian',
+      :operatingsystem        => 'ubuntu',
+      :operatingsystemrelease => '14.04',
+      :lsbdistcodename        => 'trusty',
+      :id                     => 'root',
+      :concat_basedir         => '/dne',
+      :path                   => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin',
+      :kernel                 => 'Linux',
+    }
+  end
+
+  it { is_expected.to compile }
+  it { is_expected.to contain_class('minecraft::source') }
+  it 'runs with defaults' do
+  end
+  context 'with defaults' do
```

Comment:
since the precondition will already include this class, you could only override this class's parameters in the precondition.  *Maybe* it'd be better to put these tests into a spec file that tests the base class.